frontend/src/pages/GroupPage.jsx

```diff
@@ -1,15 +1,152 @@
+    }, []);
+
     return (
         <>
             <Template
                 showSearch={true}
                 showProfile={true}
-                body={<h1>Group Page</h1>}
+                body={
+                    <div className="d-flex flex-column left-side">
+                        <ModalHeader title={`Hi Group ${group}`} />
+                        <div className="d-flex flex-column align-items-left">
+                            <LeftPanel
+                                body={
+                                    <div className="flex-row">
+                                        <ModalHeader title="Members" />
+                                        <div className="d-flex row agenda">
+                                            {members.map((member) => (
+                                                <h3>{member.username}</h3>
+                                            ))}
+                                        </div>
+                                        <div className="flex-row">
+                                            <Button
+                                                title="ADD MEMBER"
+                                                type="primary"
+                                                clicked={() =>
+                                                    setButtonAddMember(true)
+                                                }
+                                            />
+                                        </div>
+                                        <div className="flex-row">
+                                            <Button
+                                                title="EDIT PROFILE"
+                                                type="secondary"
+                                                clicked={() =>
+                                                    setButtonGroupProfile(true)
+                                                }
+                                            />
+                                        </div>
+                                    </div>
+                                }
+                            />
+                        </div>
+                        <div className="d-flex flex-column right-side-top">
+                            <SectionTitle title="Our Group" />
+                            <div className="agenda big">{groupDescription}</div>
+                        </div>
+                        <div className="d-flex right-side-middle">
+                            <SectionTitle title="Our Notes" />
+                            <div className="d-flex flex-row right-side-down">
+                                <Button
+                                    title="UPLOAD"
+                                    type="primary"
+                                    clicked={() => setButtonUpload(true)}
+                                />
+                            </div>
+                        </div>
```

Comment:
Not necessarily in this PR, but I still think we should use flex box for the layout of the personal and group page. You can take a look at how I formatted the View Note Page: just make flex-columns for the left and right side and set the width for one of those columns.

Reply:
If you aren't doing this in this PR, can you create another issue to do so in the backlog?

Reply:
Added in the backlog.